src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int RADIUS = 10;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(4); // 4 different type of figures
+        switch (figureType) {
+            case 0:
+                return new Rectangle(random.nextDouble(10),
+                        random.nextDouble(10), colorSupplier.getRandomColor());
```

Comment:
```suggestion
                        random.nextDouble(10), randomColor);
```

same for each figure

Reply:
bot then all "random figures" would be have same color